core/src/main/scala/com/raphtory/internals/context/RaphtoryContext.scala

```diff
@@ -1,48 +1,100 @@
 package com.raphtory.internals.context
 
 import cats.effect.IO
-import com.oblac.nomen.Nomen
+import cats.effect.Resource
 import com.raphtory.api.analysis.graphview.DeployedTemporalGraph
-import com.raphtory.internals.management._
+import com.raphtory._
+import com.raphtory.internals.communication.repositories.LocalTopicRepository
+import cats.effect.unsafe.implicits.global
+import com.raphtory.internals.management.Prometheus
+import com.raphtory.internals.management.QuerySender
+import com.raphtory.internals.management.Scheduler
+import com.raphtory.protocol.RaphtoryService
 import com.typesafe.config.Config
 import com.typesafe.scalalogging.Logger
 import org.slf4j.LoggerFactory
+import com.raphtory.internals.components.querymanager.Query
+import com.raphtory.internals.context.GraphException._
 
-import scala.collection.mutable
+class RaphtoryContext(serviceAsResource: Resource[IO, RaphtoryService[IO]], config: Config) {
+  protected val logger: Logger = Logger(LoggerFactory.getLogger(this.getClass))
 
-abstract class RaphtoryContext {
+  // With this API user is trying to connect to an existing graph with following expectations:
+  // - if the graph is not available with the provided graph id, throw an exception otherwise fetch the graph to work with
+  def runWithGraph[T](graphID: String, destroy: Boolean = false)(f: DeployedTemporalGraph => T): T = {
```

Comment:
these two functions have a lot in common, I'd prefer an idempotent operation callig `runWithGraph` but behave like `runWithNewGraphOrCreate` instead of failing if the graph is not there we create it
we can add params if we absolutely must fail

Reply:
We have already visited this with @miratepuffin. We don't want creating a new graph as default behaviour.